File: python-amberum-taf/page_objects/login_page.py
import os

from selenium.webdriver.common.by import By
from selenium.webdriver.support import expected_conditions as ec
from selenium.webdriver.support.ui import WebDriverWait


class LoginPage:

    def __init__(self, driver):
        self.driver = driver

    def login(self):
        login_box = WebDriverWait(self.driver, 10).until(
            ec.element_to_be_clickable((By.ID, "login"))
        )

        login_box.send_keys(os.environ["LOGIN_USER"])
        self.driver.find_element(By.ID, "password").send_keys(os.environ["LOGIN_PASSWORD"])
        self.driver.find_element(By.ID, "doLogin").click()

        menu = WebDriverWait(self.driver, 5).until(
            ec.visibility_of_element_located((By.ID, "side-menu"))
        )

File: java-amberum-taf/src/test/java/com/amberum/pages/login/LoginPage.java
package com.amberum.pages.login;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

public class LoginPage {

    private WebDriver driver;

    public LoginPage(WebDriver driver) {
        this.driver = driver;
    }

    public void login() {
        new WebDriverWait(driver, 20).until(ExpectedConditions.urlContains("/login"));

        driver.findElement(By.id("login")).sendKeys(System.getenv("LOGIN_USER"));
        driver.findElement(By.id("password")).sendKeys(System.getenv("LOGIN_PASSWORD"));
        driver.findElement(By.id("doLogin")).click();

        WebElement menu = driver.findElement(By.id("side-menu"));
        new WebDriverWait(driver, 5).until(ExpectedConditions.visibilityOf(menu));
    }

}
